state-reconstruct-fetcher/src/types/v3.rs

```diff
@@ -141,9 +148,113 @@ fn parse_pubdata_from_calldata(
 
     // Parse compressed state diffs.
     // NOTE: Is this correct? Ignoring the last 32 bytes?
-    let end_point = bytes.len() - 32;
-    let mut state_diffs = parse_compressed_state_diffs(&bytes[..end_point], pointer)?;
+    let diff_bytes = if shorten {
+        let end_point = bytes.len() - 32;
+        &bytes[..end_point]
+    } else {
+        bytes
+    };
+    let mut state_diffs = parse_compressed_state_diffs(diff_bytes, pointer)?;
     l2_to_l1_pubdata.append(&mut state_diffs);
 
     Ok(l2_to_l1_pubdata)
 }
+
+async fn parse_pubdata_from_blobs(
+    bytes: &[u8],
+    pointer: &mut usize,
+    client: &reqwest::Client,
+    blobs_url: &str,
+) -> Result<Vec<L2ToL1Pubdata>, ParseError> {
+    let mut l = bytes.len() - *pointer;
+    let mut blobs = Vec::new();
+    while *pointer < l {
+        let pubdata_commitment = &bytes[*pointer..*pointer + PUBDATA_COMMITMENT_SIZE];
+        let blob = get_blob(&pubdata_commitment[48..96], client, blobs_url).await?;
+        let mut blob_bytes = ethereum_4844_data_into_zksync_pubdata(&blob);
+        blobs.append(&mut blob_bytes);
+        *pointer += PUBDATA_COMMITMENT_SIZE;
+    }
+
+    l = blobs.len();
+    while l > 0 && blobs[l - 1] == 0u8 {
+        l -= 1;
+    }
+
+    let blobs_view = &blobs[..l];
+    let mut pointer = 0;
```

Comment:
Frankly I don't think it ever made much sense to have functions taking explicit pointer - but since it's already implemented, I'm not keen to rewrite the code just to change the style...

Reply:
Yeah, no need to refactor it now since we'll probably end up rewriting a large chunk of this code in the future. Though, I'd recommend leaving some notes where this can be done so we don't forget to.